### PR TITLE
fix: Delete unused resources

### PR DESCRIPTION
**Pull request by mimotej**
This PR deletes various unused resources, which cause kubeval to fail and there is no reason to keep them in the repository.
Part of: https://github.com/operate-first/SRE/issues/107

### CODE REVIEW

File: kubeflow-components/overlays/osc/osc-cl1/kustomization.yaml [hunk 1]
```diff
@@ -1,5 +0,0 @@
-apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (HumairAK) [inline kubeflow-components/overlays/osc/osc-cl1/kustomization.yaml:1]: This is actually deployed on the `osc-cl1` cluster but not managed by ArgoCD. I think it's fine to remove the `kubeflow-components` dir, but can you move the kfdef this is referencing base, to: https://github.com/operate-first/apps/tree/master/kfdefs/overlays/osc/osc-cl1 under a new directory called `kubeflow` and include it in this kustomization as well: https://github.com/operate-first/apps/blob/master/kfdefs/overlays/osc/osc-cl1/kustomization.yaml

File: kfdefs/overlays/osc/osc-cl1/kubeflow/kustomization.yaml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 ---
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
-namespace: kubeflow
```
Comment (HumairAK) [inline kfdefs/overlays/osc/osc-cl1/kubeflow/kustomization.yaml:4]: keep this

File: kubeflow-components/overlays/osc/osc-cl1/kustomization.yaml (hunk 1 above)
Comment (mimotej) [reply kubeflow-components/overlays/osc/osc-cl1/kustomization.yaml:1]: Moved kfdef as requested :+1: .

File: kfdefs/overlays/osc/osc-cl1/kubeflow/kustomization.yaml (hunk 2 above)
Comment (mimotej) [reply kfdefs/overlays/osc/osc-cl1/kubeflow/kustomization.yaml]: Changed :+1:

**Comment by HumairAK** [comment]
/lgtm

@4n4nd can you also take a look

**Review by tumido — COMMENTED** [review]
:raised_hands:  love it :heart:

**Review by 4n4nd — APPROVED** [review]
/lgtm

**Pull request merged**